extensions/oidc/runtime/src/main/java/io/quarkus/oidc/runtime/CodeAuthenticationMechanism.java

```diff
@@ -96,10 +96,14 @@ private static QuarkusSecurityIdentity augmentIdentity(SecurityIdentity security
 
     private CompletionStage<SecurityIdentity> performCodeFlow(IdentityProviderManager identityProviderManager,
             RoutingContext context) {
-        CompletableFuture<SecurityIdentity> cf = new CompletableFuture<>();
         JsonObject params = new JsonObject();
 
-        params.put("code", context.request().getParam("code"));
+        String code = context.request().getParam("code");
+        if (code == null) {
```

Comment:
@stuartwdouglas Might make sense to log at this point that the IDP has not completed the user authentication

Reply:
you could add some debug logging I guess, but it should not be logged at any level higher.